### PR TITLE
[Gen4] Fix gen4 occidentary wrong values (#165)

### PR DESCRIPTION
**Pull request by Real96**
Fix for #165

### CODE REVIEW

File: Source/Core/Gen4/Generators/WildGenerator4.cpp [hunk 1]
```diff
@@ -26,9 +26,10 @@
 #include <Core/Util/EncounterSlot.hpp>
 
 WildGenerator4::WildGenerator4(u32 initialAdvances, u32 maxAdvances, u16 tid, u16 sid, u8 genderRatio, Method method,
-                               const StateFilter &filter) :
+                               const StateFilter &filter, bool isPlatinumVersion) :
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Generators/WildGenerator4.cpp:29]: I find it difficult that platinum is the only game that is different. Are you sure?

**Review by Admiral-Fish — COMMENTED** [review]
In the future please have each unrelated change be a separate pull request. This is really a PR with 3 unrelated changes. It is easy to approve the things that look okay when they are separate.

File: Source/Core/Gen4/Generators/WildGenerator4.cpp (hunk 1 above)
Comment (Real96) [reply Source/Core/Gen4/Generators/WildGenerator4.cpp:29]: Yes i made many tests

**Comment by Real96** [comment]
> In the future please have each unrelated change be a separate pull request. This is really a PR with 3 unrelated changes. It is easy to approve the things that look okay when they are separate.

Sorry you're right t.t

**Comment by Real96** [comment]
> In the future please have each unrelated change be a separate pull request. This is really a PR with 3 unrelated changes. It is easy to approve the things that look okay when they are separate.

I separated the prs. I deleted RockSmash commit cause it wasn't the correct fix, i'll wait for you test first

File: Source/Core/Gen4/Generators/WildGenerator4.cpp (hunk 1 above)
Comment (Real96) [reply Source/Core/Gen4/Generators/WildGenerator4.cpp:29]: > I find it difficult that platinum is the only game that is different. Are you sure?

If you check the commit, the only different game isn't Platinum, but Diamond/Pearl. It make sense, cause Diamond and Pearl were released first, so things changed with Pt and HGSS.

Comment (Admiral-Fish) [reply Source/Core/Gen4/Generators/WildGenerator4.cpp:29]: Can you change the name to just `platinum`

File: Source/Core/Gen4/Generators/WildGenerator4.cpp [hunk 2]
```diff
@@ -100,6 +101,15 @@ std::vector<WildState4> WildGenerator4::generateMethodJ(u32 seed) const
...
             state.setLevel(encounterArea.calcLevel(state.getEncounterSlot()));
+            if (isPlatinumVersion)
+            {
+                occidentary += 2; // Add 2 to compensate for the game's advances after the battle ends
+            }
+            else
+            {
+                occidentary += 6; // Add 6 to compensate for the game's advances after the battle ends
+            }
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Generators/WildGenerator4.cpp:112]: occidentary += platinum ? 2 : 6;

File: Source/Core/Gen4/Generators/WildGenerator4.cpp [hunk 3]
```diff
@@ -109,6 +119,15 @@ std::vector<WildState4> WildGenerator4::generateMethodJ(u32 seed) const
...
             state.setLevel(encounterArea.calcLevel(state.getEncounterSlot(), go.nextUShort(occidentary)));
+            if (isPlatinumVersion)
+            {
+                occidentary += 2; // Add 2 to compensate for the game's advances after the battle ends
+            }
+            else
+            {
+                occidentary += 6; // Add 6 to compensate for the game's advances after the battle ends
+            }
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Generators/WildGenerator4.cpp:130]: occidentary += platinum ? 2 : 6;

File: Source/Core/Gen4/Generators/WildGenerator4.cpp [hunk 4]
```diff
@@ -125,6 +144,15 @@ std::vector<WildState4> WildGenerator4::generateMethodJ(u32 seed) const
...
             state.setLevel(encounterArea.calcLevel(state.getEncounterSlot(), go.nextUShort(occidentary)));
+            if (isPlatinumVersion)
+            {
+                occidentary += 3; // Add 3 to compensate for the game's advances after the battle ends
+            }
+            else
+            {
+                occidentary += 7; // Add 7 to compensate for the game's advances after the battle ends
+            }
+
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Generators/WildGenerator4.cpp:155]: occidentary += platinum ? 3 : 7;

File: Source/Core/Gen4/Generators/WildGenerator4.hpp [hunk 5]
```diff
@@ -38,6 +39,7 @@ class WildGenerator4 : public WildGenerator
     std::vector<WildState4> generateChainedShiny(u32 seed) const;
 
     EncounterArea4 encounterArea;
+    bool isPlatinumVersion;
```
Comment (Admiral-Fish) [inline Source/Core/Gen4/Generators/WildGenerator4.hpp:42]: Change to `platinum`

**Review by Admiral-Fish — CHANGES_REQUESTED** [review]

File: Source/Core/Gen4/Generators/WildGenerator4.cpp (hunk 2 above)
Comment (Real96) [reply Source/Core/Gen4/Generators/WildGenerator4.cpp]: done

File: Source/Core/Gen4/Generators/WildGenerator4.cpp (hunk 3 above)
Comment (Real96) [reply Source/Core/Gen4/Generators/WildGenerator4.cpp]: done

File: Source/Core/Gen4/Generators/WildGenerator4.cpp (hunk 4 above)
Comment (Real96) [reply Source/Core/Gen4/Generators/WildGenerator4.cpp]: done

File: Source/Core/Gen4/Generators/WildGenerator4.hpp (hunk 5 above)
Comment (Real96) [reply Source/Core/Gen4/Generators/WildGenerator4.hpp]: done

**Pull request merged**